fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -520,10 +544,16 @@ private SqlQueryData processReferenceParm(Class<?> resourceType, QueryParameter
 
         boolean parmValueProcessed = false;
         for (QueryParameterValue value : queryParm.getValues()) {
-            // Handle query parm representing this name/value pair construct:
-            // <code> {name} = {resource-type/resource-id}</code>
+            String targetResourceType = null;
             searchValue = SqlParameterEncoder.encode(value.getValueString());
-
+            
+            // Make sure we split out the resource type if it is included in the search value
+            String[] parts = value.getValueString().split("/");
+            if (parts.length == 2) {
+                targetResourceType = parts[0];
+                searchValue = parts[1];
+            }
```

Comment:
Would this ever be a URL reference?  e.g. http://localhost/fhir/api/v4/Patient/1

Reply:
It passes the reference search unit tests e.g. 
```
        assertSearchReturnsSavedResource("Reference-relative", "https://example.com/Patient/123");
```